src/json.hpp

```diff
@@ -2217,17 +2219,17 @@ class lexer
         }
     }
 
-    static void strtof(float& f, const char* str, char** endptr) noexcept
+    static void strtof(float& f, const value_t* str, value_t** endptr) noexcept
```

Comment:
Since `std::strtof` takes `const char *`, this will cause a failure if `value_t` is not `char`.
The same for `std::strtod` and `std::strtold`.